### PR TITLE
Add DumpCompactionLog tool and other minor change

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
This PR adds a DumpCompactionLog tool to parse binary compaction log file.
It also adds some minor change to blob store.

### CODE REVIEW

**Comment by zzmao** [comment]
Looks like some tests failing.

File: ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java [hunk 1]
```diff
@@ -380,6 +385,35 @@ byte to indicate whether startOffsetOfLastIndexSegmentForDeleteCheck is present
     }
   }
 
+  @Override
+  public String toString() {
+    StringBuffer buffer = new StringBuffer();
```
Comment (jsjtzyy) [inline ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java:390]: I think you wanted `StringBuilder` here.

File: ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java [hunk 2]
```diff
@@ -479,5 +513,26 @@ safeToken if not null (see StoreFindToken#toBytes())
       bufWrap.put(safeTokenBytes);
       return buf;
     }
+
+    @Override
+    public String toString() {
+      StringBuffer buffer = new StringBuffer();
```
Comment (jsjtzyy) [inline ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java:519]: Same here

File: ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java [hunk 3]
```diff
@@ -479,5 +513,26 @@ safeToken if not null (see StoreFindToken#toBytes())
...
+      buffer.append("CompactionDetal:")
```
Comment (jsjtzyy) [inline ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java:520]: typo: CompactionDetails

File: ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java [hunk 4]
```diff
@@ -479,5 +513,26 @@ safeToken if not null (see StoreFindToken#toBytes())
...
+          .append("\n")
+          .append("CommitStartTime:")
+          .append(commitStartTime)
+          .append("\n")
+          .append("CleanupStartTime:")
+          .append(cleanupStartTime)
+          .append("\n")
+          .append("CycleEndTime:")
+          .append(cycleEndTime)
+          .append("\n");
```
Comment (jsjtzyy) [inline ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java:534]: Probably we also need to check if token exists and append it to string if present.

File: ambry-tools/src/main/java/com/github/ambry/store/DumpCompactionLog.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.tools.util.ToolUtils;
+import com.github.ambry.utils.SystemTime;
+import com.github.ambry.utils.Time;
+import com.github.ambry.utils.Utils;
+import java.io.File;
+
+
+public class DumpCompactionLog {
```
Comment (jsjtzyy) [inline ambry-tools/src/main/java/com/github/ambry/store/DumpCompactionLog.java:29]: nit: call this `DumpCompactionLogTool`

**Review by jsjtzyy — COMMENTED** [review]
LGTM after minor comments are addressed and tests succeed.

**Review by zzmao — APPROVED** [review]
Please fix the test error.

File: ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java (hunk 1 above)
Comment (justinlin-linkedin) [reply ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java:390]: That's right, string builder should be much faster.

File: ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java (hunk 2 above)
Comment (justinlin-linkedin) [reply ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java:519]: updated

**Review by jsjtzyy — APPROVED** [review]

**Pull request merged**